src/any_llm/providers/mistral/utils.py

```
@@ -417,15 +433,76 @@ def _create_openai_moderation_response_from_mistral(
     Patches messages for Mistral API compatibility.
 
+    - Rebuilds assistant thinking blocks so reasoning traces survive multi-turn replay.
     - Inserts an assistant message with "OK" content between a tool message and a user message.
     - Validates the message sequence to ensure correctness.
     """
     processed_msg: list[dict[str, Any]] = []
     for i, msg in enumerate(messages):
+        if msg.get("role") == "assistant":
+            rebuilt_content = _build_mistral_assistant_content(msg)
+            if rebuilt_content is not None:
+                msg = {**msg, "content": rebuilt_content}
```

Ruff (0.16.1)

[warning] `for` loop variable `msg` overwritten by assignment target

(PLW2901)